cmd/kubeadm/app/cmd/phases/init/kubelet.go

```diff
@@ -76,6 +77,13 @@ func runKubeletStart(c workflow.RunData) error {
 		return errors.Wrap(err, "error writing a dynamic environment file for the kubelet")
 	}
 
+	// Write the instance kubelet configuration file to disk.
+	if features.Enabled(data.Cfg().FeatureGates, features.NodeLocalCRISocket) {
+		if err := kubeletphase.WriteInstanceConfigToDisk(data.Cfg().NodeRegistration.CRISocket, data.KubeletDir()); err != nil {
```

Comment:
are we calling this on `kubeadm join` too?

Reply:
also added.